fix(client/plugins): add keyboard interaction trap to all plugin modals

barmac opened the pull request:
Closes #1692

nikku commented on client/src/plugins/camunda-plugin/start-instance-tool/StartInstanceTool.js:
```diff
@@ -13,7 +13,7 @@ import React, { PureComponent } from 'react';
 
 import CamundaAPI from '../shared/CamundaAPI';
-import KeyboardInteractionTrap from '../shared/KeyboardInteractionTrap';
+import KeyboardInteractionTrap from '../../../app/primitives/modal/KeyboardInteractionTrap';
```
Can't we add the trap as a default to _all_ modals? It feels odd to me that every new modal has to add this for itself.

barmac replied:
I was reluctant to do that because then we need to pass `triggerAction` to the modals. Yet, we can try this out.

nikku commented:
Maybe that could be made available by the app via `React.Context`?

barmac commented:
I'll try that out.

barmac commented:
I've implemented the context-based solution and extracted the test refactoring to a separate PR: https://github.com/camunda/camunda-modeler/pull/1822

Please feel free to review now.

oguzeroglu commented on client/src/app/__tests__/AppSpec.js:
```diff
@@ -2643,6 +2645,12 @@ function createApp(options = {}, mountFn=shallow) {
   let app;
 
+  const flags = options.flags || {
+    [ DISABLE_REMOTE_INTERACTION ]: true
```
Why do we need this?

barmac replied:
Without the flag we open privacy preferences in App tests :joy:

oguzeroglu replied:
🐵

barmac replied:
![image](https://user-images.githubusercontent.com/28307541/83754065-a3eab180-a66b-11ea-856a-68d81273c1dd.png)

oguzeroglu commented on client/src/app/App.js:
```diff
@@ -1809,155 +1810,159 @@ export class App extends PureComponent {
         <div className={ css.App }>
 
-          <SlotFillRoot>
+          <KeyboardInteractionTrapContext.Provider value={ this.triggerAction }>
```
Maybe I'm missing the point here but is there a reason why we don't only wrap PluginsRoot component? AFAIK KeyboardInteractionTrap was meant to be used by plugins.

barmac replied:
It's meant to be used by modals, regardless of whether they are embedded in plugins or not. The benefit of the `KeyboardInteractionTrap` is that you don't accidentally activate editor's shortcuts while a modal is open. Also, there is still one modal which has not been moved to the plugins yet: KeyboardShortcuts. Therefore, I don't see reason to limit the scope of the context.

nikku commented on client/src/app/primitives/modal/Modal.js:
```diff
@@ -60,14 +61,16 @@ export default class Modal extends PureComponent {
-          <div className="modal-content">
-            { children }
-            { onClose && (<Close onClick={ this.close } />) }
+      <KeyboardInteractionTrap>
```
Really good solution! Thanks :clap:

barmac replied:
Thanks :)
